app/src/main/java/com/example/myapplication/MainActivity.kt

```diff
@@ -1,17 +1,125 @@
 package com.example.myapplication
 
+import android.content.Intent
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.ImageButton
+import android.widget.ImageView
+import android.widget.TextView
+import android.widget.Toast
 import androidx.appcompat.app.AppCompatActivity
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
 
-class MainActivity : AppCompatActivity()  {
-/*Create new Account
-    Correct Input values
-        Password (8 charachteers/digits)
-        Email Format
-    Email already used - Error
-    Successful registration (new user is present in CredManager)
-    Case Insensitive in email
-
-    Apply it to SignUpDetailsActivity
-    After successful registration, go to CreateAccount activity
-    */
+class MainActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        val sharedPreferences = getSharedPreferences("user_prefs", MODE_PRIVATE)
+        val isLoggedIn = sharedPreferences.getBoolean("is_logged_in", false)
+
+        if (isLoggedIn) {
+            loadMainContent()
+        } else {
+            if (savedInstanceState == null) {
```

Comment:
I don't think existence of savedInstanceState implies existence of the view
Did you check otherwise?